freestyle/shared/src/main/scala/freestyle/free.scala

```diff
@@ -16,49 +16,54 @@
 
 package freestyle
 
-import scala.annotation.{compileTimeOnly, StaticAnnotation}
-import scala.language.experimental.macros
-import scala.reflect.macros.blackbox
+import scala.reflect.macros.blackbox.Context
 
-trait FreeModuleLike {
-  type Op[A]
-}
+object freeImpl {
 
-@compileTimeOnly("enable macro paradise to expand @free macro annotations")
-class free extends StaticAnnotation {
-  def macroTransform(annottees: Any*): Any = macro free.impl
-}
+  object messages {
+
+    val invalid = "Invalid use of the `@free` annotation"
 
-object free {
+    val abstractOnly = "The `@free` macro annotation can only be applied to either a trait or an abstract class."
 
-  def impl(c: blackbox.Context)(annottees: c.Expr[Any]*): c.universe.Tree = {
+    val noCompanion = "The trait or class annotated with `@free` must have no companion object."
```

Comment:
This is a limitation that we may want to remove if it's easy to just add the generated code into an existing companion object.